### PR TITLE
tsify instructionsets

### PR DESCRIPTION
**Pull request by partouf**
TLDR: Problem fixed, don't name your properties `default`

I'm getting this error locally and I'm clueless on what's going on:
```
$ npm run test

> compiler-explorer@0.0.3 test
> mocha -b


/opt/compiler-explorer/cemain/lib/instructionsets.ts:26
    default;
           ^

SyntaxError: Invalid or unexpected token
    at Object.m._compile (/opt/compiler-explorer/cemain/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
```

### CODE REVIEW

**Comment by partouf** [comment]
Github test results seems to have some "better" information

https://github.com/compiler-explorer/compiler-explorer/runs/8219172982?check_suite_focus=true#step:5:22
```
/home/runner/work/compiler-explorer/compiler-explorer/lib/instructionsets.ts:25
import _ from'underscore';export class InstructionSets{default;supported;constructor(){...
```

Obviously it's superweird that it would internally emit "default;supported;", that doesn't make any sense...
Although I have nothing to compare it to, as these intermediate minified things are not saved to disk.. I think?

**Comment by partouf** [comment]
`npm run ts-check` and `npm run ts-compile` just .. work... so 

AH wait, I now see that ts-compile also emits something similar:
```
import _ from 'underscore';
export class InstructionSets {
    default;
    supported;
    constructor() {
```

But if I look at s3-handler, that has the same `variablename;` pattern in the compiled versions.

Could this just be mocha thing, where all of a sudden it starts to be bothered by things?

**Comment by partouf** [comment]
OH

I guess default is a keyword in Javascript !

**Comment by partouf** [comment]
We don't actually have unittests for this class, so I don't know if it still works after removing underscore. Maybe should add some...

File: lib/instructionsets.ts [hunk 1]
```diff
@@ -80,27 +84,29 @@ export class InstructionSets {
...
-    async getCompilerInstructionSetHint(compilerArch, exe) {
+    async getCompilerInstructionSetHint(compilerArch: string | boolean, exe: string): Promise<string> {
         return new Promise(resolve => {
-            if (compilerArch) {
-                _.each(this.supported, (method, instructionSet) => {
-                    for (let target of method.target) {
+            if (compilerArch && typeof compilerArch === 'string') {
+                for (const instructionSet in this.supported) {
+                    const method = this.supported[instructionSet];
+                    for (const target of method.target) {
```
Comment (RubenRBS) [inline lib/instructionsets.ts:92]: Cant comment on lines not edited:

Can you add returns after the resolves (94 & 103)? Although the promise status won't get updated after the first resolve/reject is issued, code still runs afaik - so we can save some loops if we stop when not needed

**Review by RubenRBS — APPROVED** [review]

File: lib/instructionsets.ts (hunk 1 above)
Comment (partouf) [reply lib/instructionsets.ts:92]: hmm, I thought it would. I remember the linter complaining when I put code after a resolve. But will try

**Pull request merged**